src/libraries/System.Text.Json/src/System/Text/Json/Serialization/WriteStack.cs

```diff
@@ -127,12 +128,14 @@ public void Push()
             {
                 if (_count == 0)
                 {
-                    // The first stack frame is held in Current.
+                    // Performance optimization: reuse the first stackframe on the first push operation.
+                    // NB need to be careful when making writes to Current _before_ the first `Push`
+                    // operation is performed.
                     _count = 1;
                 }
                 else
                 {
-                    JsonTypeInfo jsonTypeInfo = Current.GetPolymorphicJsonPropertyInfo().JsonTypeInfo;
+                    JsonTypeInfo jsonTypeInfo = Current.GetNestedJsonTypeInfo();
```

Comment:
nit: could this be `GetPolymorphicJsonTypeInfo()`?

Reply:
The existing name isn't accurate, it's not necessarily getting a polymorphic JsonTypeInfo, it is resolving the right JsonTypeInfo for a nested value to be serialized (property or collection element) which may or may not be polymorphic.